client/go/outline/parse.go

```diff
@@ -59,31 +65,47 @@ func doParseTunnelConfig(input string) *InvokeMethodResult {
 			}
 		}
 
-		// Process provider error, if present.
-		if tunnelConfig.Error != nil {
-			platErr := &platerrors.PlatformError{
-				Code:    platerrors.ProviderError,
-				Message: tunnelConfig.Error.Message,
+		if hasKey(yamlValue, "transport") || hasKey(yamlValue, "error") {
+			// New format. Parse as tunnel config
+			tunnelConfig := parseTunnelConfigRequest{}
+			if err := yaml.Unmarshal([]byte(input), &tunnelConfig); err != nil {
+				return &InvokeMethodResult{
+					Error: &platerrors.PlatformError{
+						Code:    platerrors.InvalidConfig,
+						Message: fmt.Sprintf("failed to parse: %s", err),
+					},
+				}
 			}
-			if tunnelConfig.Error.Details != "" {
-				platErr.Details = map[string]any{
-					"details": tunnelConfig.Error.Details,
+
+			// Process provider error, if present.
+			if tunnelConfig.Error != nil {
+				platErr := &platerrors.PlatformError{
+					Code:    platerrors.ProviderError,
+					Message: tunnelConfig.Error.Message,
+				}
+				if tunnelConfig.Error.Details != "" {
+					platErr.Details = map[string]any{
+						"details": tunnelConfig.Error.Details,
+					}
 				}
+				return &InvokeMethodResult{Error: platErr}
 			}
-			return &InvokeMethodResult{Error: platErr}
-		}
 
-		// Extract transport config as an opaque string.
-		transportConfigBytes, err := yaml.Marshal(tunnelConfig.Transport)
-		if err != nil {
-			return &InvokeMethodResult{
-				Error: &platerrors.PlatformError{
-					Code:    platerrors.InvalidConfig,
-					Message: fmt.Sprintf("failed normalize config: %s", err),
-				},
+			// Extract transport config as an opaque string.
+			transportConfigBytes, err := yaml.Marshal(tunnelConfig.Transport)
+			if err != nil {
+				return &InvokeMethodResult{
+					Error: &platerrors.PlatformError{
+						Code:    platerrors.InvalidConfig,
+						Message: fmt.Sprintf("failed normalize config: %s", err),
+					},
```

Comment:
*nit* We can use `Cause` here to include the `err`, so we don't need to format it in `Message`.

Reply:
Just `Cause: platerrors.ToPlatformError(err)`